### PR TITLE
Enhanced finetune.py with custom loss weights and reference energies

### PR DESCRIPTION
**Pull request by timduignan**
- Updated finetune.py to support:
  - Custom loss weights (energy, forces, stress)
  - Custom reference energies (fixed or trainable)
  - Automatic handling of conservative vs direct models
  - Loading reference energies from JSON or text files

- Added comprehensive FINETUNING_GUIDE.md with:
  - Data format requirements (ASE SQLite database)
  - Step-by-step instructions for data preparation
  - Examples for converting various data formats
  - Usage examples for all features

- Added example_reference_energies.json as a reference template

This enables finetuning with different reference energy schemes (e.g., ORCA, VASP, custom DFT methods) while maintaining compatibility with pretrained Orb models.

### CODE REVIEW

File: finetune.py [hunk 1]
```diff
@@ -221,6 +223,180 @@ def build_train_loader(
...
+        
+        # Make trainable if requested
+        if args.trainable_reference_energies:
+            logging.info("Making custom reference energies trainable...")
+            model.heads["energy"].reference.linear.weight.requires_grad = True
+        else:
+            logging.info("Custom reference energies are FIXED (not trainable)")
+            model.heads["energy"].reference.linear.weight.requires_grad = False
+        logging.info("=" * 60)
+
```
Comment (vsimkus) [inline finetune.py]: This finetuning script is generally intended as a demo on how one can do finetuning, but people should generally write their own finetuning scripts, since we cannot cover all the use cases. 

Therefore I think it would be much better to have a flag in `orb_v3_conservative_inf_omat` and others which sets `requires_grad=True` for the reference energies, similar to the `train` flag. This approach is much more "pythonic" and hints to people who are not using this finetuning script, that if they want to finetune reference energies, they should change that flag.

Could you perhaps rebase on my branch, where I've implemented this already (https://github.com/orbital-materials/orb-models/pull/132), or otherwise add this to you PR?

File: finetune.py [hunk 2]
```diff
@@ -221,6 +223,180 @@ def build_train_loader(
...
+        ref_params = model.heads['energy'].reference.linear.weight.numel()
+        logging.info(f"  Added {ref_params} trainable reference energy parameters")
+        
+        # Show some example values
+        ref_weights = model.heads['energy'].reference.linear.weight.data.squeeze()
+        logging.info(f"  Current H (element 1): {ref_weights[1].item():.2f}")
+        logging.info(f"  Current C (element 6): {ref_weights[6].item():.2f}")
+        logging.info(f"  Current N (element 7): {ref_weights[7].item():.2f}")
+        logging.info(f"  Current O (element 8): {ref_weights[8].item():.2f}")
+        logging.info("=" * 60)
```
Comment (vsimkus) [inline finetune.py]: This finetuning script is generally intended as a demo on how one can do finetuning, but people should generally write their own finetuning scripts, since we cannot cover all the use cases. 

Therefore I think it would be much better to have a flag in `orb_v3_conservative_inf_omat` and others which sets `requires_grad=True` for the reference energies, similar to the `train` flag. This approach is more "pythonic" and hints to people who are not using this finetuning script, that if they want to finetune reference energies, they should change that flag.

Could you perhaps rebase on my branch, where I've implemented this already (https://github.com/orbital-materials/orb-models/pull/132), or otherwise add this to you PR?

Comment (vsimkus) [reply finetune.py]: Also, there's a duplication with lines 374-379, so I think it would be good to fix that.

File: finetune.py [hunk 3]
```diff
@@ -221,6 +223,180 @@ def build_train_loader(
...
+        if args.stress_loss_weight is not None:
+            if is_conservative:
+                model.loss_weights["grad_stress"] = args.stress_loss_weight
+                logging.info(f"  grad_stress: {args.stress_loss_weight}")
+            elif is_direct and "stress" in model.heads:
+                model.loss_weights["stress"] = args.stress_loss_weight
+                logging.info(f"  stress: {args.stress_loss_weight}")
+        
+        logging.info(f"Final loss_weights: {model.loss_weights}")
+        logging.info("=" * 60)
```
Comment (vsimkus) [inline finetune.py]: Same here, I think it's cleaner to add an argument in `orb_v3_conservative_inf_omat` (and others) called `loss_weights` that takes a dictionary and passes to the constructor of the model.

File: FINETUNING_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+- `--custom_reference_energies`: Path to reference energies file (JSON or text format)
+- `--trainable_reference_energies`: Make reference energies trainable during finetuning
+
+### Model Types
+
+The script automatically handles the differences between conservative and direct models:
+
+- **Conservative models** (e.g., `orb_v3_conservative_omol`):
+  - Use `grad_forces` and `grad_stress` keys
```
Comment (vsimkus) [inline FINETUNING_GUIDE.md]: This is unclear on its own. I think this is supposed to say "loss-weight keys"

File: finetune.py (hunk 2 above)
Comment (tgmaxson) [reply finetune.py]: > This finetuning script is generally intended as a demo on how one can do finetuning, but people should generally write their own finetuning scripts, since we cannot cover all the use cases.
> 
> Therefore I think it would be much better to have a flag in `orb_v3_conservative_inf_omat` and others which sets `requires_grad=True` for the reference energies, similar to the `train` flag. This approach is more "pythonic" and hints to people who are not using this finetuning script, that if they want to finetune reference energies, they should change that flag.
> 
> Could you perhaps rebase on my branch, where I've implemented this already (#132), or otherwise add this to you PR?

I believe I am the user that inspired Tim to submit this PR.  I want to give a user perspective for this.

My honest opinion is that the finetuning script is too rudimentary as is, beyond simple "use cases".  The simple LR is potentially ok, but right now there are a few gotchas with the current fine tuning script such as the lack of reference energies and lack of any error logging to the console.

One other complaint I have with the finetuning scripts is that I always have to remove wandb from them as its not an optional import.  The fine tuning works quite well even in a simple form, but I strongly believe user adoption is driven not by the potential of a method but instead the ease of adoption of the method.  

I have had some experience working with the fine tuning script as an MLIP user / workflow person, but with very little PyTorch experience directly. I am happy to talk to someone directly in the near future if my perspective would be useful to Orb.
